src/Server/TCPHandler.cpp

```diff
@@ -845,6 +889,15 @@ void TCPHandler::receiveQuery()
     state.is_empty = false;
     readStringBinary(state.query_id, *in);
 
+    if (hello_credentials)
+    {
+        connection_context.setUser(hello_credentials.user, hello_credentials.password, socket().peerAddress());
+        query_context->setUser(hello_credentials.user, hello_credentials.password, socket().peerAddress());
```

Comment:
I don't like this place for now, i.e. calling setUser twice, but this is because of how packages are received right now (loop in the receivePacket), this can be fixed (by parsing Cluster package out of the receivePacket loop, although this will not look nice as-is right now, or by replace Hello package with Hellov2/ExtendedHello as was suggested, but I don't like this idea since it depends on exceptions and it is a little bit harder to test), but is this looks really bad or it is just me?

Reply:
Double call of `setUser()` does look pretty bad for me. What if we move copying `connection_context` to `query_context` into this function?

I like the idea about Hellov2 more because it allows to keep the protocol easy (well, except the case when the server is older than the client). And I think it's better for the future because we can add to Hellov2 some extra field of a variable length so we won't ever have to implement Hellov3.

Reply:
>Double call of setUser() looks pretty bad for me. What if we move copying connection_context to query_context into this function?

Hm, not sure that copying the whole context is better, but I'm ok with this (since once LDAP will be integrated setUser may take more time)

>I like the idea about Hellov2 more because it allows to keep the protocol easy (well, except the case when the server is older than the client).

I like it, but I don't like complexity in the protocol, not sure that clean protocol should have this...
Although I don't have strong feeling about this (I guess), so you are voting for extending Hello instead of separate Cluster package (like right now) ?

Reply:
>Hm, not sure that copying the whole context is better, but I'm ok with this (since once LDAP will be integrated setUser may take more time)

Hm, it also may copy the query_id (not a problem, but AFAFICS the connection context should not have any query_id)